crates/rpc/rpc/src/eth/api/call.rs

```diff
@@ -320,15 +323,16 @@ where
             )?;
         };
 
-        // pick a point that's close to the estimated gas
+        // Pick a point that's close to the estimated gas
         let mut mid_gas_limit = std::cmp::min(
             gas_used * 3,
```

Comment:
Here as we have potentially done in the meantime a new simulation with the `optimistic_gas_limit` we should not rather use `res.result.gas_used()` which can be different from `gas_used` in the case where the condition `optimistic_gas_limit < highest_gas_limit` was verified previously.

Reply:
@mattsse What do you think about this?

Reply:
not following tbh

Reply:
I just wanted to say that here the `gas_used` variable used for the calculation is still the old variable defined above with `let gas_used = res.result.gas_used();`. But between this declaration and the use of `gas_used` here, in the case where `if optimistic_gas_limit < highest_gas_limit` is true, a new execution is done with `(res, env) = self.transact(&mut db, env)?;` . As in this case `res` is updated, `res.result.gas_used()` may be different from its old value before this re-execution.

So I was wondering if on this line it wasn't better to put `res.result.gas_used() * 3` directly, wdyt?

Reply:
Fixed https://github.com/paradigmxyz/reth/pull/8541